src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -596,7 +614,25 @@ public void apply(VariantContext vc, ReadsContext readsContext, ReferenceContext
 
             if (!failedJexlMatch &&
                     (!selectRandomFraction || Utils.getRandomGenerator().nextDouble() < fractionRandom)) {
-                vcfWriter.add(filteredGenotypeToNocall);
+                //remove annotations being dropped and write variantcontext
+                final VariantContextBuilder rmAnnotationsBuilder = new VariantContextBuilder(filteredGenotypeToNocall);
+                for (String infoField : infoAnnotationsToDrop) {
+                    rmAnnotationsBuilder.rmAttribute(infoField);
+                }
+
+                final ArrayList<Genotype> genotypesToWrite = new ArrayList<>();
+                for (Genotype genotype : filteredGenotypeToNocall.getGenotypes()) {
+                    final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(genotype).noAttributes();
+                    final Map<String, Object> attributes = new HashMap<>(genotype.getExtendedAttributes());
+                    for (String genotypeAnnotation : genotypeAnnotationsToDrop) {
+                        attributes.remove(genotypeAnnotation);
+                    }
+                    genotypeBuilder.attributes(attributes);
+                    genotypesToWrite.add(genotypeBuilder.make());
+                }
+                rmAnnotationsBuilder.genotypes(GenotypesContext.create(genotypesToWrite));
+                final VariantContext variantContextToWrite = rmAnnotationsBuilder.make();
+                vcfWriter.add(variantContextToWrite);
```

Comment:
Looks good, except I just realized that after this PR, we'll always recreate the entire VC and all the genotypes, even in the common case where there are no dropped fields. It might be worth moving this code out to a separate method, and only making the actual changes necessary, (i.e., if nothing is dropped, just return the input vc, and if only info fields are dropped, avoid recreating all the genotypes).